flink-core/src/main/java/org/apache/flink/api/common/io/DelimitedInputFormat.java

```diff
@@ -755,10 +759,13 @@ public void reopen(FileInputSplit split, Long state) throws IOException {
 				fillBuffer(0);
 			} else {
 				this.splitLength = this.splitStart + split.getLength() - this.offset;
-				if (splitLength <= 0) {
-					this.end = true;
-				}
 			}
 		}
+
+		initializeSplit(split, state);
```

Comment:
do we need to initialize a split in case it has ended?

Reply:
True, it doesn't hurt but shouldn't be necessary